src/entity.cpp

```diff
@@ -7610,16 +7610,24 @@ void Entity::monsterAddNearbyItemToInventory(Stat* myStats, int rangeToFind, int
 				{
 					//Drop that item out of the monster's inventory, and add this item to the monster's inventory.
 					Item* itemToDrop = static_cast<Item*>(replaceInventoryItem->element);
-					if ( itemToDrop && myStats->type != AUTOMATON ) // Automatons are hoarders.
+					if ( itemToDrop )
 					{
-						dropItemMonster(itemToDrop, this, myStats, itemToDrop->count);
+						if ( !(myStats->type == AUTOMATON && list_Size(&myStats->inventory) < maxInventoryItems) )
```

Comment:
If there's no else statement, shouldn't this be combined with the previous if statement?